Modules/rat_board.py

```diff
@@ -199,10 +199,50 @@ def find_by_uuid(self, guid: UUID) -> Optional[Rescue]:
 
         """
         for rescue in self.rescues.values():
-            if rescue.case_id == guid:
+            if rescue.uuid == guid:
                 return rescue
         return None
 
+    def search(self, case: Union[str, int, UUID]) -> Optional[Rescue]:
```

Comment:
I believe this signature to be too broadly defined. This makes the very defensive programming, which unknown pointed out, necessary. Having to type check your arguments a lot is often an indicator of a badly designed function / method signature, even in Python.

For getting rescues directly by an int (board index) or by a UUID we already have the appropriate methods. This method's purpose, therefore, should not overlap with that of the others.

More specifically, this method should, in my mind, only take a string to then figure out what it represents, convert it and pass it to the appropriate method.

Reply:
That makes sense to me. The purpose of the function is to allow the commands to "search" for the appropriate rescue and it'll always pass a string anyway. I'll refactor.